### PR TITLE
Adding parser support for AT TIME ZONE

### PR DESCRIPTION
**Pull request by hash-16**
### Description

Added a sql_dialect to engine grammer in order to hit sys catalog instead of pg_catalog function.

Extension PR :- [Link](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1822)
 
### Issues Resolved

[List any issues this PR will resolve]
 
### Check List

- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/backend/parser/gram.y [hunk 1]
```diff
@@ -14539,10 +14539,20 @@ a_expr:		c_expr									{ $$ = $1; }
 				}
 			| a_expr AT TIME ZONE a_expr			%prec AT
 				{
+					if(sql_dialect == SQL_DIALECT_TSQL)
```
Comment (KushaalShroff) [inline src/backend/parser/gram.y:14542]: Why can't we add this support in ANTLR?

**Comment by KushaalShroff** [comment]
Please add an appropriate description to your PRs

File: src/backend/parser/gram.y (hunk 1 above)
Comment (hash-16) [reply src/backend/parser/gram.y:14542]: I tried doing the change but it seem a little more then expected as per the work I did so tried using engine code .

File: src/backend/parser/gram.y [hunk 2]
```diff
@@ -14539,10 +14539,20 @@ a_expr:		c_expr									{ $$ = $1; }
...
+    				{
+	    				$$ = (Node *) makeFuncCall(list_make2(makeString("sys"), makeString("timezone")),
```
Comment (surendravishnoi) [inline src/backend/parser/gram.y]: please take care of indentation

Comment (hash-16) [reply src/backend/parser/gram.y]: Addressed

**Review by forestkeeper — DISMISSED** [review]

**Review by surendravishnoi — DISMISSED** [review]

**Review by surendravishnoi — APPROVED** [review]

**Review by forestkeeper — APPROVED** [review]

**Pull request merged**